app/Controllers/Discussions/ImageController.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace App\Controllers\Discussions;
+
+use App\Controllers\BaseController;
+use App\Models\ImageModel;
+use ReflectionException;
+
+/**
+ * Class Image
+ */
+class ImageController extends BaseController
+{
+    /**
+     * Upload image and return uploaded file URL.
+     *
+     * @throws ReflectionException
+     */
+    public function upload()
+    {
+        if (! $this->policy->can('images.upload')) {
+            return $this->response->setStatusCode(401)->setJSON([
+                'error' => 'You are not allowed to upload images.',
+            ]);
+        }
+
+        if (! $this->validate([
+            'image' => [
+                'uploaded[image]', 'max_size[image,2048]',
```

Comment:
I think we are going to want the image size to be configurable at some point down the road, but we can worry about that later.

Reply:
Will do!